xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/BaseModal.java

```diff
@@ -69,6 +80,42 @@ public void close()
         waitForClosed();
     }
 
+    public void validateWCAG(WCAGContext wcagContext, boolean checkCache, String cssSelector) {
+        if (!wcagContext.isWCAGEnabled()) {
+            // Block WCAG validation if it is not enabled, in all cases.
+            return;
+        }
+
+        try {
+            long startTime = System.currentTimeMillis();
+            // Run WCAG tests on the current UI page if the current URL + PO class name are not in the cache, or if checking
+            // the cache is disabled.
+            XWikiWebDriver driver = this.getDriver();
+            if (!checkCache || wcagContext.isNotCached(driver.getCurrentUrl(), this.getClass().getName())) {
+                AxeBuilder axeBuilder = wcagContext.getAxeBuilder();
+                // Only analyze the modal itself
+                axeBuilder.include(cssSelector);
+                Results axeResult = axeBuilder.analyze(driver);
+                wcagContext.addWCAGResults(driver.getCurrentUrl(), this.getClass().getName(), axeResult);
+                long stopTime = System.currentTimeMillis();
+                long deltaTime = stopTime - startTime;
+                LOGGER.info("[{} : {}] WCAG Validation on this element took [{}] ms.",
+                    driver.getCurrentUrl(), this.getClass().getName(), deltaTime);
+                wcagContext.addWCAGTime(deltaTime);
+            } else {
+                // If the identifying pair is already in the cache, don't perform accessibility validation.
+                LOGGER.debug("[{} : {}] This combination of URL:class was already WCAG-checked.",
+                    driver.getCurrentUrl(), this.getClass().getName());
+            }
+        } catch (Exception e) {
+            if (wcagContext.shouldWCAGStopOnError()) {
+                throw e;
+            } else {
+                LOGGER.debug("Error during WCAG execution, but ignored thanks to wcagStopOnError flag: ", e);
```

Comment:
Better be an info than a debug here no? We don't enable debug often, and it could be useful to still see those, or could that be an issue on the CI?

Reply:
This is in line with https://github.com/xwiki/xwiki-platform/blob/0ddc451a08d512f0b092bfe9923f0e42b6e62967/xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/BasePage.java#L713